Added note about format, and light editing (#2013)
timescale/docs: timescaledb/tutorials/grafana/create-dashboard-and-panel.md

# Creating a Grafana dashboard and panel

Grafana is organized into `Dashboards` and `Panels`. A dashboard represents a

> view onto the performance of a system, and each dashboard consists of one or

Reviewer: ```suggestion
view into the performance of a system, and each dashboard consists of one or
```

more panels, which represents information about a specific metric related to
that system.

In this tutorial, you'll build a simple dashboard, connect it to TimescaleDB,
and visualize data.

## Prerequisites

Before you begin, make sure you have:

*   [Installed TimescaleDB][install-timescale].
*   [Set up Grafana][install-grafana].

When your installation of TimescaleDB and Grafana are complete, ingest the data
found in the [NYC Taxi Cab][nyc-taxi] tutorial and configure Grafana to connect
to that database.

## Build a new dashboard

Start by creating a new dashboard. In the far left of the Grafana user
interface, you'll see a `+` icon. If you hover over it, you'll see a `Create`
menu, within which is a `Dashboard` option. Select that `Dashboard` option.

After creating a new dashboard, you'll see a `New Panel` screen, with options
for `Add Query` and `Choose Visualization`. In the future, if you already have a

> dashboard with panels, you can click on the `+` icon at the top of the

Author: ```suggestion
dashboard with panels, you can click the `+` icon at the top of the
```

Grafana user interface, which enables you to add a panel to an existing
dashboard.

To proceed with the tutorial, add a new visualization by clicking the `Choose
Visualization` option.

At this point, you'll have several options for different Grafana visualizations.
Choose the first option, the `Graph` visualization.

<img class="main-content__illustration" src="https://assets.iobeam.com/images/docs/screenshots-for-grafana-tutorial/grafana_visualizations.png" alt="Grafana visualizations to choose from"/>

There are multiple ways to configure the panel, but you can accept all the
defaults and create a simple `Lines` graph.

In the far left section of the Grafana user interface, select the 'Queries' tab.

<img class="main-content__illustration" src="https://assets.iobeam.com/images/docs/screenshots-for-grafana-tutorial/create_grafana_query.png" alt="How to create a new Grafana query"/>

Instead of using the Grafana query builder, edit the query directly. In the

> view, click on the `Edit SQL` button at the bottom.

Author: ```suggestion
view, click the `Edit SQL` button at the bottom.
```

<img class="main-content__illustration" src="https://assets.iobeam.com/images/docs/screenshots-for-grafana-tutorial/edit_sql_in_grafana.png" alt="Edit custom SQL queries in Grafana"/>

> Before you can begin authoring our query, you also want to set the query database

Author: ```suggestion
Before you can begin authoring your query, you also want to set the query database
```

> to the New York City taxi cab datasource you connected to earlier:

Author: ```suggestion
to the New York City taxi cab dataset you connected to earlier:
```

<img class="main-content__illustration" src="https://assets.iobeam.com/images/docs/screenshots-for-grafana-tutorial/set_data_source.png" alt="Switching data sources in Grafana"/>

<highlight type="note">
If you are visualizing time series data in Grafana, make sure you select
`Time series` from the `Format As` drop down in the query builder.
</highlight>

### Visualize metrics stored in TimescaleDB

> Let's start by creating a visualization that answers the question "How many

Author: ```suggestion
Start by creating a visualization that answers the question "How many
```

ORDER BY day;
```

You need to alter this query to support Grafana's unique query syntax.

#### Modifying the SELECT statement

First, modify the `date_trunc` function to use the TimescaleDB `time_bucket`
function. You can consult the TimescaleDB
[API Reference on time_bucket][time-bucket-reference]
for more information on how to use it properly.

Take a look at the `SELECT` portion of this query. First, bucket the results
into one day groupings using the `time_bucket` function. If you set the `Format`
of a Grafana panel to be `Time series`, for use in the graph panel for example,

> then the query must return a column named `time` that returns either a SQL

Author: ```suggestion
then the query must return a column named `time` that returns either an SQL
```

FROM rides
```

#### The Grafana timeFilter function

> Grafana time-series panels include a tool thatlets you filter on a given time

Reviewer: ```suggestion
Grafana time-series panels include a tool that lets you filter on a given time
```
